File: src/main/java/com/rs/java/game/player/actions/skills/crafting/spinningwheel/SpinningProduct.java
package com.rs.java.game.player.actions.skills.crafting.spinningwheel;

import com.rs.kotlin.Rscm;
import com.rs.java.game.player.actions.skills.crafting.leather.ReqItem;

public final class SpinningProduct {

    private final Object id;
    private final int level;
    private final double xp;
    private final ReqItem[] requirements;

    public SpinningProduct(Object id, int level, double xp, ReqItem... requirements) {
        this.id = id;
        this.level = level;
        this.xp = xp;
        this.requirements = requirements;
    }

    public int getId() {
        if (id instanceof Integer)
            return (Integer) id;
        return Rscm.lookup((String) id);
    }

    public int getLevel() { return level; }
    public double getXp() { return xp; }
    public ReqItem[] getRequirements() { return requirements; }
}

File: src/main/java/com/rs/java/game/player/actions/skills/crafting/spinningwheel/SpinningProduct.kt
package com.rs.java.game.player.actions.skills.crafting.spinningwheel

import com.rs.java.game.player.actions.skills.crafting.ReqItem
import com.rs.kotlin.rscm.Rscm

class SpinningProduct(
    private val idRef: Any,
    val level: Int,
    val xp: Double,
    vararg val requirements: ReqItem,
) {
    fun getId(): Int = idRef as? Int ?: Rscm.lookup(idRef as String)
}
